unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -138,6 +138,15 @@ public int numChars() {
   }
 
   /**
+   * Returns a 64-bit integer that can be used as the prefix used in sorting.
+   */
+  public long getPrefix() {
+    long p = PlatformDependent.UNSAFE.getLong(base, offset);
+    p = java.lang.Long.reverseBytes(p);
```

Comment:
cc @tellison can you take a look at this (or find somebody else) to see if we can fix it by just not doing the reverse bytes in big-endian?